Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -759,8 +759,17 @@ def network_shutdown(self):
             self.threadpool.cancel_all_pending_tasks()
             self.threadpool = None
 
+    def save_download_pstate_callback(self, resume_data, infohash, pstate):
+        pstate.set('state', 'engineresumedata', resume_data)
+        self.save_download_pstate(infohash, pstate)
+
     def save_download_pstate(self, infohash, pstate):
         """ Called by network thread """
+        resume_data = pstate.get('state', 'engineresumedata')
+        if isinstance(resume_data, Deferred):
```

Comment:
you don't do anything until you've got all you need. If you need more help, please join the IRC channel